src/NServiceBus.Metrics.AcceptanceTests/Tests/QueueLength/When_publishing_message.cs

```diff
@@ -38,26 +37,43 @@ public class When_publishing_message : NServiceBusAcceptanceTest
                     await session.Subscribe<TestEventMessage1>();
                     await session.Subscribe<TestEventMessage2>();
                 }))
-                .Done(c => c.Headers1.Count == 2 && c.Headers2.Count == 2)
+                .WithEndpoint<MonitoringSpy>()
+                .Done(c => c.Headers1.Count == 2 && c.Headers2.Count == 2 && SequencesReported(c))
                 .Run()
                 .ConfigureAwait(false);
+        }
 
-            var sessionIds = new[] { AssertHeaders(context.Headers1), AssertHeaders(context.Headers2) };
+        static bool SequencesReported(Context context)
```

Comment:
Can you have a look at https://github.com/Particular/NServiceBus.Metrics/pull/35#discussion_r125588411?